app/src/main/java/com/amaze/filemanager/fragments/MainFragment.java

```diff
@@ -1027,8 +1027,8 @@ public void loadlist(final String path, final boolean back, final OpenMode openM
             loadFilesListTask.cancel(true);
         }
 
-        loadFilesListTask = new LoadFilesListTask(ma.getActivity(), path, ma, openMode,(data) -> {
-            if (data.second != null) {
+        loadFilesListTask = new LoadFilesListTask(ma.getActivity(), path, ma, openMode, (data) -> {
+            if (data != null && data.second != null) {
```

Comment:
When is data null?

Reply:
i was just about to add this log.. I copied all of the crashes and this was one of them. It happened when trying to copy multiple files but I'm not 100% sure how to reproduce.

```
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime     java.lang.NullPointerException: Attempt to read from field 'java.lang.Object android.support.v4.util.Pair.second' on a null object reference
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at com.amaze.filemanager.fragments.MainFragment.lambda$loadlist$3(MainFragment.java:1031)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at com.amaze.filemanager.fragments.MainFragment$$Lambda$4.onAsyncTaskFinished(Unknown Source)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.onPostExecute(LoadFilesListTask.java:215)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.onPostExecute(LoadFilesListTask.java:65)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at android.os.AsyncTask.finish(AsyncTask.java:660)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at android.os.AsyncTask.-wrap1(AsyncTask.java)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:677)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at android.os.Handler.dispatchMessage(Handler.java:102)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at android.os.Looper.loop(Looper.java:154)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at android.app.ActivityThread.main(ActivityThread.java:6642)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at java.lang.reflect.Method.invoke(Native Method)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
12-01 09:50:12.791    com.amaze.filemanager    26528    -    E    AndroidRuntime         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
```

Reply:
btw - I would agree w/you that it'd be better to understand how this could ever be null.. I just was working through issues so I could copy files

Reply:
Here's the culprit, I think: [`if(isCancelled()) return null;`](https://github.com/TeamAmaze/AmazeFileManager/blob/37ea7763d7861a5a57e8a90a6d6278c077e795eb/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java#L100).

Reply:
Or, more probably, [lines 107 to 125](https://github.com/TeamAmaze/AmazeFileManager/blob/37ea7763d7861a5a57e8a90a6d6278c077e795eb/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java#L107-L125). When SMB fails it returns null; also, I might be blind :D